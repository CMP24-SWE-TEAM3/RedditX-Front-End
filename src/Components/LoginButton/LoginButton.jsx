import React, {useContext} from 'react';
import {LoginStyled} from "./LoginButton.styled";
import AuthContext from "Contexts/auth-context";

/**
 * Component  that displays the login button
 * @returns {Component.React} returns a login button component
 */
const LoginButton = () => {
    /**
     * ctx is the context  for the login button component that will be disappeared after the login button has been clicked
     * to show navigation in authenticated state
     *
     * @type {{onLogin: function(), onLogout: function(), isLoggedIn: boolean}}
     */
    const ctx = useContext(AuthContext);
    return (
        <>
            <LoginStyled id={'login-button-from-HomePage'} onClick={() => {
                ctx.onLogin()
            }} variant="outline-primary rounded-pill h-4">Log In</LoginStyled>
        </>

    );
};

export default LoginButton;
