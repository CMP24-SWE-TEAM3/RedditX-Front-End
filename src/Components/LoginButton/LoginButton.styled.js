import styled from "styled-components";
import {Button} from "react-bootstrap";

export const LoginStyled = styled(Button)`
  color: ${({theme})=>theme.background.primary};
  border-color: ${({theme})=>theme.color.secondary};
  background-color: ${({theme})=>theme.color.secondary};
  width: 120px;
  margin: 15px;
  height: 32px;
  padding:0 10px;
  &:hover {
    background-color: #1484d6 !important;
  }
  &:active {
    background-color: #0079d3 !important;
  }

  font-weight:bold;
`