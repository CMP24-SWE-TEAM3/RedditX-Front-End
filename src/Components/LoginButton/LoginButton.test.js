import {shallow} from 'enzyme';

// import Components
import LoginButton from "./LoginButton";

describe("Login Button Component", () => {
    it("this is a test for Login Button", () => {
        expect(shallow(<LoginButton/>)).toMatchSnapshot();

    });
});