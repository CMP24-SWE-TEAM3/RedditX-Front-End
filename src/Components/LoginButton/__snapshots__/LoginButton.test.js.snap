// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Button Component this is a test for Login Button 1`] = `ShallowWrapper {}`;
